astropy/cosmology/core.py

```diff
@@ -345,12 +345,16 @@ def __eq__(self, other: Any, /) -> bool:
         if other.__class__ is not self.__class__:
             return NotImplemented  # allows other.__eq__
 
-        # check all parameters in 'other' match those in 'self'
-        equivalent = self.__equiv__(other)
+        # check all parameters in 'other' match those in 'self' and 'other' has
+        # no extra parameters (latter part should never happen b/c same class)
+        # TODO! element-wise when there are array cosmologies
+        params_eq = (set(self.__all_parameters__) == set(other.__all_parameters__)
+                     and all(np.all(getattr(self, k) == getattr(other, k))
+                             for k in self.__all_parameters__))
         # non-Parameter checks: name
         name_eq = (self.name == other.name)
 
-        return equivalent and name_eq
+        return params_eq and name_eq
```

Comment:
Currently `name_eq` is always evaluated even if `params_eq` is already known to be `False`. This could be avoided with
```python
return params_eq and self.name == other.name
```
But perhaps it would be even better to compare the names first?